src/models/state/wallet/wallet_state.rs

```diff
@@ -664,25 +720,42 @@ impl WalletState {
     /// Scan the given transaction for announced UTXOs as recognized by owned
     /// `SpendingKey`s, and then verify those announced UTXOs are actually
     /// present.
-    fn scan_for_announced_utxos<'a>(
+    fn scan_for_utxos_announced_to_known_keys<'a>(
         &'a self,
         tx_kernel: &'a TransactionKernel,
     ) -> impl Iterator<Item = IncomingUtxo> + 'a {
         // scan for announced utxos for every known key of every key type.
-        self.get_all_known_spending_keys()
-            .flat_map(|key| key.scan_for_announced_utxos(tx_kernel))
+        self.get_all_known_spending_keys().flat_map(|key| key.scan_for_announced_utxos(tx_kernel))
+
+        // filter for presence in transaction
+        //
+        // note: this is a nice sanity check, but probably is un-necessary
+        //       work that can eventually be removed.
+        .filter(|au| match tx_kernel.outputs.contains(&au.addition_record()) {
+            true => true,
+            false => {
+                warn!("Transaction does not contain announced UTXO encrypted to own receiving address. Announced UTXO was: {:#?}", au.utxo);
+                false
+            }
+        })
+    }
```

Comment:
I don't understand this comment, that we don't actually need to check for the presence of the announced UTXO. Really? I kind of like this check. I guess it depends on the function contract you want for this method. Does it guarantee that the `IncomingUtxo`s reported are actually present in the block. If so, the comment is incorrect.

I don't think you added the comment but it shows up in the diff, so now is as good a time as any to discuss it.